### PR TITLE
Add some pre-calculated metrics

### PR DESCRIPTION
**Pull request by morningman**
1. max io util of disks
2. max network send/receive bytes rate of all network devices
3. base/cumulative compaction request counter and failure counter

### CODE REVIEW

File: be/src/util/system_metrics.cpp [hunk 1]
```diff
@@ -435,4 +435,65 @@ void SystemMetrics::_update_fd_metrics() {
     fclose(fp);
 }
 
+int64_t SystemMetrics::get_max_io_util(
+        const std::map<std::string, int64_t>& lst_value, int64_t interval_sec) {
+    int64_t max = 0;
+    for (auto& it : _disk_metrics) {
+        int64_t cur = it.second->io_time_ms.value();
+        const auto& find = lst_value.find(it.first);
```
Comment (imay) [inline be/src/util/system_metrics.cpp]: find return a temporary variable, do not use reference

File: be/src/util/system_metrics.h [hunk 2]
```diff
@@ -40,6 +41,19 @@ class SystemMetrics {
     // update metrics
     void update();
 
+    void get_disks_io_time(std::map<std::string, int64_t>& map);
+    int64_t get_max_io_util(
+        const std::map<std::string, int64_t>& lst_value, int64_t interval_sec);
+
+    void get_network_traffic(
```
Comment (imay) [inline be/src/util/system_metrics.h]: output variable use pointer

File: be/src/util/system_metrics.cpp [hunk 3]
```diff
@@ -435,4 +435,65 @@ void SystemMetrics::_update_fd_metrics() {
...
+            continue;
+        }
+        int64_t incr = cur - find->second;
+        if (incr > max) max = incr;
+    }
+    return max / interval_sec / 10;
+}
+
+void SystemMetrics::get_disks_io_time(std::map<std::string, int64_t>& map) {
+    map.clear();
```
Comment (imay) [inline be/src/util/system_metrics.cpp]: I think that caller clear is better

File: be/src/common/daemon.cpp [hunk 4]
```diff
@@ -111,17 +112,27 @@ void* memory_maintenance_thread(void* dummy) {
...
+ * 4. max network send bytes rate
+ * 5. max network receive bytes rate
  */
 void* calculate_metrics(void* dummy) {
     int64_t last_ts = -1L;
     int64_t lst_push_bytes = -1;
     int64_t lst_query_bytes = -1;
+    std::map<std::string, int64_t> lst_disks_io_time;
+    std::map<std::string, int64_t> lst_net_send_bytes;
+    std::map<std::string, int64_t> lst_net_receive_bytes;
```
Comment (imay) [inline be/src/common/daemon.cpp]: you can put these map in `while` block to avoid clear

**Review by imay — CHANGES_REQUESTED** [review]

File: be/src/util/system_metrics.cpp (hunk 1 above)
Comment (morningman) [reply be/src/util/system_metrics.cpp]: ok

File: be/src/util/system_metrics.h (hunk 2 above)
Comment (morningman) [reply be/src/util/system_metrics.h]: there is not output variables in parameter list

File: be/src/util/system_metrics.cpp (hunk 3 above)
Comment (morningman) [reply be/src/util/system_metrics.cpp]: ok

File: be/src/common/daemon.cpp (hunk 4 above)
Comment (morningman) [reply be/src/common/daemon.cpp]: The values in these maps need to be used at next loop, so they can not be put in the while block.

**Review by imay — APPROVED** [review]

**Pull request merged**